centinel/primitives/tls.py

```diff
@@ -22,6 +23,11 @@ def get_fingerprint(host, port=443, external=None):
         tls_error = str(exp)
         exception = exp
 
+    # this comes out as unicode, but m2crypto breaks if it gets
+    # something other than a string, so convert to ascii
+    if type(cert) == unicode or test:
+        cert = cert.encode('ascii', 'ignore')
```

Comment:
This shouldn't be anything other than ascii text anyway. The whole thing is a hack because m2crypto doesn't support unicode strings (which it should). We are getting the encoded cert, so it can't be anything other than ascii and if it is, we will get an error when we try to process it.